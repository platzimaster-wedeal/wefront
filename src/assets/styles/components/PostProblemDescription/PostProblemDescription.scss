@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.post-problem-description {
  width: 100%;
  margin: $space-size-extra-small 0;
  height: auto;

  & strong {
    @include font-size($font-size-normal);
    font-weight: 500;
    color: $color-black-90;
  }

  & p {
    padding: $space-size-extra-small 0;
    line-height: $space-size-big;
    @include font-size($font-size-small);

    @include media(tablet) {
      @include font-size($font-size-normal);
    }
  }
}
