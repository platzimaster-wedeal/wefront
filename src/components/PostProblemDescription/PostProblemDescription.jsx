import React from "react";

// Styles
import "../../assets/styles/components/PostProblemDescription/PostProblemDescription.scss";

const PostProblemDescription = ({ description }) => {
 return (
  <section className="post-problem-description">
   <strong className="post-problem-description__title">Description</strong>
   <p className="post-problem-description__content">
    {description ||
     "Here must be some long description of the problem, thank you, next!"}
   </p>
  </section>
 );
};

export default PostProblemDescription;
