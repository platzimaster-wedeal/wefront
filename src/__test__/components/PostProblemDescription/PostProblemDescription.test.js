import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostProblemDescription from "../../../components/PostProblemDescription/PostProblemDescription";

describe("Test for the component PostProblemDescription :D", () => {
 const PostProblemDescriptionMounted = mount(<PostProblemDescription />);
 const PostProblemDescriptionCreated = create(<PostProblemDescription />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostProblemDescriptionCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostProblemDescriptionMounted.length).toEqual(1);
  });
  test("Should the description of the post be a string", () => {
   expect(
    typeof PostProblemDescriptionMounted.find(
     ".post-problem-description__content"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the post must be a string", () => {
   expect(
    typeof PostProblemDescriptionMounted.find(
     ".post-problem-description__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the post be Description", () => {
   expect(
    PostProblemDescriptionMounted.find(
     ".post-problem-description__title"
    ).text()
   ).toEqual("Description");
  });
 });
});
