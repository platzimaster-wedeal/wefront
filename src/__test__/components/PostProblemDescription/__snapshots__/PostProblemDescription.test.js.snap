// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostProblemDescription :D Visual tests for the component should the comopnent have the same tree 1`] = `
<section
  className="post-problem-description"
>
  <strong
    className="post-problem-description__title"
  >
    Description
  </strong>
  <p
    className="post-problem-description__content"
  >
    Here must be some long description of the problem, thank you, next!
  </p>
</section>
`;
